src/unix/core.c

```diff
@@ -1041,6 +1043,55 @@ int uv_getrusage(uv_rusage_t* rusage) {
 }
 
 
+int uv_getrusage(uv_rusage_t* rusage) {
+  return uv__getrusage(RUSAGE_SELF, rusage);
+}
+
+
+int uv_getrusage_thread(uv_rusage_t* rusage) {
+#if defined(__APPLE__)
+  thread_basic_info_data_t info;
+  thread_t thread = mach_thread_self();
+  mach_msg_type_number_t count = THREAD_BASIC_INFO_COUNT;
+  kern_return_t kr = thread_info(thread,
+                                 THREAD_BASIC_INFO,
+                                 (thread_info_t)&info,
+                                 &count);
+  if (kr != KERN_SUCCESS) {
+    mach_port_deallocate(mach_task_self(), thread);
+    return UV_EINVAL;
+  }
+
+  rusage->ru_utime.tv_sec = info.user_time.seconds;
+  rusage->ru_utime.tv_usec = info.user_time.microseconds;
+  rusage->ru_stime.tv_sec = info.system_time.seconds;
+  rusage->ru_stime.tv_usec = info.system_time.microseconds;
+
+  /* Since thread_info doesn't provide these fields, set them to 0 */
+  rusage->ru_maxrss = 0;
+  rusage->ru_ixrss = 0;
+  rusage->ru_idrss = 0;
+  rusage->ru_isrss = 0;
+  rusage->ru_minflt = 0;
+  rusage->ru_majflt = 0;
+  rusage->ru_nswap = 0;
+  rusage->ru_inblock = 0;
+  rusage->ru_oublock = 0;
+  rusage->ru_msgsnd = 0;
+  rusage->ru_msgrcv = 0;
+  rusage->ru_nsignals = 0;
```

Comment:
What about initializing `rusage` to 0 using `memset()` as it's done on the Windows part to avoid this whole block?

Reply:
True that!